compute/managed-instances/demo/main.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from os import path
+from requests import get
+from re import sub
+from socket import gethostname
+from socketserver import ThreadingMixIn
+
+PORT_NUMBER = 80
+HTML_TEMPLATES_PATH = path.dirname(path.realpath(__file__)) + '/templates'
+HEALTHY = True
+
+env = Environment(
+    loader=FileSystemLoader(HTML_TEMPLATES_PATH),
+    autoescape=select_autoescape('html')
+)
+
+
+class request_handler(BaseHTTPRequestHandler):
+    def get_zone(self):
+        r = get('http://metadata.google.internal/'
+                'computeMetadata/v1/instance/zone',
+                headers={'Metadata-Flavor': 'Google'})
+        if r.status_code == 200:
+            return sub(r'.+zones/(.+)', r'\1', r.text)
+        else:
+            return ''
```

Comment:
Are there any failure cases here that we should be throwing errors on instead of returning the empty string? This seems like unexpected behavior.

Reply:
Unsuccessfully getting the zone is not a critical failure.  The zone (and the template) are fetched simply for informational purposes on the UI.  The demo still works perfectly fine if both fetches fail.  If I were to put something in the failure case, it would be an error message along the lines of 'zone not found'/'template not found'.